### ISSUE TITLE
look-up-table doesn't works.

### ISSUE THREAD

**Issue by kanryu** [halide/Halide#2667]
Hi,

I write a filter with a lut.

https://gist.github.com/kanryu/f66dd3371c220367ffdf9c964bcc8141

But the content of lut is not changed regardless of the Params(b_sigma, c_sigma, g_sigma) and remains fixed.
Lut was created with the initial value given to Param and it seems that it can not be rewritten later.

How can we solve this problem?

**Comment by zvookin** [comment]
Are you calling get_luminorImpl each time those parameters change? The lut
is being realized in the construction routine, thus the computed values are
put into the filter as constants at compile time.

If you want the lut to vary without recompiling each time, you will need to
make the sigmas and such Halide Params and then schedule the lut
compute_root. If the values are reused frequently, using memoize may be
worthwhile. (Or the lut itself can be passed to the compiled filter as an
argument.)

-Z-

**Comment by kanryu** [comment]
> If you want the lut to vary without recompiling each time, you will need to
make the sigmas and such Halide Params and then schedule the lut
compute_root. If the values are reused frequently, using memoize may be
worthwhile. (Or the lut itself can be passed to the compiled filter as an
argument.)

I made it!
https://gist.github.com/kanryu/d176b9480e2e15b2b68a16814588a6ba

That version runs 4.3 times faster than before.

and I found that Func::memoize(), Func::in() methods are not defined.
